lib/models/builder.js

```diff
@@ -33,19 +33,35 @@ function vizEnabled() {
   return !!process.env.BROCCOLI_VIZ;
 }
 
-function outputViz(count, result) {
+function computeVizInfo(count, result, resultAnnotation) {
   var nodes = result.graph.__heimdall__.toJSONSubgraph();
 
-  fs.writeJsonSync('broccoli-viz.' + (count) + '.json', {
+  var vizJson = {
     /* we should calculate the summary more generically */
     summary: {
-      buildCount: count,
+      build: {
+        type: resultAnnotation.type,
+        count: count,
+        outputChangedFiles: result.outputChanges,
+      },
       output: result.directory,
       totalTime: result.totalTime,
       totalNodes: nodes.length,
     },
     nodes: nodes,
-  });
+  };
+
+  if (resultAnnotation.type === 'rebuild') {
+    vizJson.summary.build.primaryFile = resultAnnotation.primaryFile;
+    vizJson.summary.build.changedFileCount = resultAnnotation.changedFiles.length;
+    vizJson.summary.build.changedFiles = resultAnnotation.changedFiles.slice(0, 10);
```

Comment:
In some cases (like a recrawl, or git branch change, of debase) it may be A very very large list (like every single file, which could be thousands or hundreds of thousands)